### ISSUE TITLE
Can s3s3mirror accept a list of files to copy?

### ISSUE THREAD

**Issue by mkanthan** [cobbzilla/s3s3mirror#74]
Let's say I have: 

```
mybucketA/images/1.png
mybucketA/images/a.png
mybucketA/sounds/2.mp3
mybucketA/sounds/b.mp3
mybucketA/videos/3.mp4
mybucketA/videos/c.mp4
```

Realistically, there are 20,000 files here.

And I run some script whose job is to output the files I need for some other process and it tells me that this is what I need in bucket B:

```
mybucketB/images/1.png
mybucketB/images/a.png
mybucketB/videos/c.mp4
```

Is there a way to make s3s3mirror take in a list of these files and copy just those, currently?

**Comment by cobbzilla** [comment]
This is not currently supported, but I really like the idea. I'm marking this issue "enhancement", and hopefully I can get to it soon. For anyone who wants to fix this sooner -- the right thing to do is:
* Add a command-line option called `--files` (or `-F`). If the value of this option is a valid file, then we presume the file contains the S3 keys to copy, one per line. If it is the special value `-`, then a list of keys is read from stdin.
* When initializing, do not use a regular KeyLister, instead use a KeyLister that simply iterates the objects in the file (or stdin) and only creates KeyJobs for those keys.

It should be fairly straightforward, though not totally trivial.

**Comment by mkanthan** [comment]
Another thing you might want to take into account is - it should probably check to see that the keys actually exist in the source bucket, and maybe reject those that aren't. It shouldn't just trust the valid file.

**Comment by cobbzilla** [comment]
@mkanthan yes, good idea. perhaps another command line option to fail and exit upon missing/invalid path, versus default behavior which is to log a warning.
